### PR TITLE
feat(lint): Add lint for usage of node globals (with autofix)

### PR DESCRIPTION
**Pull request by nathanwhit**
From upgrading `deno_lint`.

Previously if you had a node project that used a bunch of node globals (`process.env`, etc), you would have to fix the errors by hand. This PR includes a new lint that detects usages of node globals (`process`, `setImmediate`, `Buffer`, etc.) and provides an autofix to import the correct value. For instance:

```ts
// main.ts
const _foo = process.env.FOO;
```

`deno lint` gives you

```ts
error[no-node-globals]: NodeJS globals are not available in Deno
 --> /home/foo.ts:1:14
  |
1 | const _foo = process.env.FOO;
  |              ^^^^^^^
  = hint: Add `import process from "node:process";`

  docs: https://lint.deno.land/rules/no-node-globals


Found 1 problem (1 fixable via --fix)
Checked 1 file
```
And `deno lint --fix` adds the import for you:

```ts
// main.ts
import process from "node:process";
const _foo = process.env.FOO;
```

### CODE REVIEW

**Comment by marvinhagemeister** [comment]
Can we update the hint message to say exactly which module to import instead of just `apropiate module`? I think most people who have only used the globals don't know that there even is a `node:process` module etc

**Comment by nathanwhit** [comment]
> Can we update the hint message to say exactly which module to import instead of just `apropiate module`? I think most people who have only used the globals don't know that there even is a `node:process` module etc

Yeah that makes sense! Opened https://github.com/denoland/deno_lint/pull/1311

**Review by marvinhagemeister — APPROVED** [review]

File: cli/tools/lint/linter.rs [hunk 1]
```diff
@@ -225,14 +226,23 @@ fn apply_lint_fixes(
...
   // remove any overlapping text changes, we'll circle
   // back for another pass to fix the remaining
   quick_fixes.sort_by_key(|change| change.range.start);
   for i in (1..quick_fixes.len()).rev() {
     let cur = &quick_fixes[i];
     let previous = &quick_fixes[i - 1];
-    let is_overlapping = cur.range.start < previous.range.end;
-    if is_overlapping {
+    // hack: deduplicate import fixes to avoid creating errors
+    if previous.new_text.trim_start().starts_with("import ") {
```
Comment (dsherret) [inline cli/tools/lint/linter.rs:238]: This doesn't seem right. We should fix this before merging.

**Review by dsherret — CHANGES_REQUESTED** [review]
Let's get rid of the hack.

File: cli/tools/lint/linter.rs (hunk 1 above)
Comment (dsherret) [reply cli/tools/lint/linter.rs:238]: Actually, I'm not sure about a way around this because for the lsp we want all the import fixes.

**Review by dsherret — APPROVED** [review]
LGTM. I can't think of a workaround for the hack for now.

**Pull request merged**